avocado/plugins/multiplex.py

```diff
@@ -115,6 +120,40 @@ def run(self, args):
             sys.exit(exit_codes.AVOCADO_ALL_OK)
 
         variants = multiplexer.MuxTree(mux_tree)
+        if args.list:
+            prefix = '  '
+            for (index, tpl) in enumerate(variants):
+                level = 0
+                if args.list_variant_id:
+                    log.info("variant: !mux")
+                    level += 1
+
+                lname = ''
+                for x in tpl:
+                    path = x.path[len('/run'):]
+                    lname += path.replace('/', '-')
+                # trim first '-'
+                lname = lname[1:]
+                if args.list_variant_id:
+                    log.info('%s%d:' % (prefix * level, index))
+                    level += 1
+                log.info('%s%s:' % (prefix * level, lname))
+                level += 1
+                env = set()
+                for node in tpl:
+                    for key, value in node.environment.iteritems():
+                        origin = node.environment_origin[key].path
+                        env.add(("%s" % key, str(value)))
+                if not env:
+                    continue
+                fmt = '%s  %%-%ds: %%s' % (prefix * level,
+                                           max([len(_[0]) for _ in env]))
+                for record in sorted(env):
+                    log.info(fmt, *record)
+                log.info("")
+
+            sys.exit(exit_codes.AVOCADO_ALL_OK)
```

Comment:
I like the addition of such a listing feature.

Still, I'll invite @ldoktor to collaborate on this, since there's a big Multiplex PR advancing (v3 is coming) which will definitely affect this.  One of the key aspects on the discussion we should have, is how abstract the concept of the variant ID should be (or in fact IS, as defined by published RFCs), and how you proposal maps to that.